src/streams/send.rs

```diff
@@ -209,10 +237,27 @@ impl StreamTx {
     ///
     /// This determines how old incoming NACKs we can reply to.
     ///
-    /// The default is 1024 packets over 3 seconds.
-    pub fn set_rtx_cache(&mut self, max_packets: usize, max_age: Duration) {
+    /// `rtx_ratio_cap` determines when to clear queued resends because of too many resends,
+    /// i.e. if `tx_sum / (rtx_sum + tx_sum) > rtx_ratio_cap`. `None` disables this functionality
+    /// so all queued resends will be sent.
+    ///
+    /// The default is 1024 packets over 3 seconds and RTX cache drop ratio of 0.15.
+    pub fn set_rtx_cache(
+        &mut self,
+        max_packets: usize,
+        max_age: Duration,
+        rtx_ratio_cap: Option<f32>,
+    ) {
         // Dump old cache to avoid having to deal with resizing logic inside the cache impl.
         self.rtx_cache = RtxCache::new(max_packets, max_age);
+        if rtx_ratio_cap.is_some() {
+            self.stats.bytes_transmitted = Some(ValueHistory::default());
```

Comment:
I am not sure why we want to reset the stats when the stream is configured. If we allow reconfiguring the rtx-cache of the stream I think the stats should not be reset.

Reply:
Well `RtxCache` is dropped, and since these stats don't do anything else i thought it would make sense, but you are probably right. Updated.